meta/store.go

```diff
@@ -416,6 +424,79 @@ func (s *Store) changeState(state raftState) error {
 	return nil
 }
 
+// monitorPeerHealth periodically checks if we have a node that can be promoted to a
+// raft peer to fill any missing slots.
+// This function runs in a separate goroutine.
+func (s *Store) monitorPeerHealth() {
+	defer s.wg.Done()
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		// Wait for next tick or timeout.
+		select {
+		case <-ticker.C:
+		case <-s.closing:
+			return
+		}
+		if err := s.promoteNodeToPeer(); err != nil {
+			s.Logger.Printf("error promoting node to raft peer: %s", err)
+		}
+	}
+}
+
+func (s *Store) promoteNodeToPeer() error {
+	// Only do this if you are the leader
+	if !s.IsLeader() {
+		return nil
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.raftState == nil {
+		return nil
+	}
+
+	peers, err := s.raftState.peers()
+	if err != nil {
+		return err
+	}
+
+	nodes := s.data.Nodes
+	var nonraft NodeInfos
+	for _, n := range nodes {
+		if contains(peers, n.Host) {
+			continue
+		}
+		nonraft = append(nonraft, n)
+	}
+
+	// Check to see if any action is required or possible
+	if len(peers) >= 3 || len(nonraft) == 0 {
+		return nil
+	}
+
+	// Sort the nodes
+	sort.Sort(nonraft)
+
+	// Get the lowest node for a deterministic outcome
+	n := nonraft[0]
+	// Set peers on the leader now to the new peers
+	if err := s.AddPeer(n.Host); err != nil {
```

Comment:
I had a LOT of logging, and looks like I deleted one line too many.  I'll add it back in right before I return so we know it was successful.